doc/running/running_nest.rst

```diff
@@ -218,16 +218,16 @@ Simulation of volume-transmitted neuromodulation in NEST can be done using "volu
                                                             "vt_ports": ["dopa_spikes"]}]})
 
 
-.. _sec-nest-third-factor-plasticity:
-
 Third-factor plasticity
 ~~~~~~~~~~~~~~~~~~~~~~~
 
-Note that when a continuous-time input port is defined in the synapse model which is connected to a postsynaptic neuron (see :ref:`Third-factor plasticity <synapses_in_nestml#third-factor-plasticity>`), a corresponding buffer is allocated in each neuron which retains the recent history of the connected state variables. This covers the most general case of different synaptic delay values and a discontinuous third-factor signal. Note that synaptic delays are in NEST interpreted as being entirely postsynaptic, that is, they correspond to the dendritic propagation delay between synapse and soma.
+When a continuous-time input port is defined in the synapse model which is connected to a postsynaptic neuron, a corresponding buffer is allocated in each neuron which retains the recent history of the needed state variables. Two options are available for how the buffer is implemented: a "continuous-time" based buffer, or a spike-based buffer (see the NEST code generator option ``continuous_state_buffering_method`` on https://nestml.readthedocs.io/en/latest/pynestml.codegeneration.html#pynestml.codegeneration.nest_code_generator.NESTCodeGenerator).
+
+By default, the "continuous-time" based buffer is selected. This covers the most general case of different synaptic delay values and a discontinuous third-factor signal. The implementation corresponds to the event-based update scheme in Fig. 4b of [Stapmanns2021]_. There, the authors observe that the storage and management of such a buffer can be expensive in terms of memory and runtime. In each time step, the value of the current dendritic current (or membrane potential, or other third factor) is appended to the buffer. The maximum length of the buffer depends on the maximum inter-spike interval of any of the presynaptic neurons.
+
+As a computationally more efficient alternative, a spike-based buffer can be selected. In this case, the third factor is not stored every timestep, but only upon the occurrence of postsynaptic (somatic) spikes. Because of the existence of a nonzero dendritic delay, the time at which the somatic spike is observed at the soma is delayed, and the time at which the third factor is sampled should match the time of the spike at the synapse, rather than the soma. When the spike-based buffering method is used, the dendritic delay is therefore ignored, because the third factor is sampled instead at the time of the somatic spike.
```

Comment:
Should it be `third factor is sampled instead at the time of the spike at the synapse` in the last line?

Reply:
The last line is correct, but the sentence before it should have read "the time at which the somatic spike is observed at the synapse is delayed", rather than "at the soma", which makes no sense. Cheers!

Reply:
The sentence is yet to be changed in the `running.rst` file.